hbase-client/src/main/java/org/apache/hadoop/hbase/client/AsyncAdmin.java

```diff
@@ -1673,4 +1682,13 @@
    * @throws IOException if a remote or network exception occurs
    */
   CompletableFuture<Void> updateRSGroupConfig(String groupName, Map<String, String> configuration);
+
+  /**
+   * Retrieve recent online records from HMaster / RegionServers.
+   * Examples include slow/large RPC logs, balancer decisions by master.
+   *
+   * @param logRequest request payload with possible filters
+   * @return Log entries representing online records from servers
+   */
+  CompletableFuture<List<LogEntry>> getLogEntries(LogRequest logRequest);
```

Comment:
Comment on admin interface has implications here. (_This needs an optional parameter to allow constrained clients to limit the size of the returned list of LogEntry._)

Reply:
Please define a limit parameter for the Admin API. I'm open to other suggestions, but by "individual use cases"  I believe you mean the RPC message classes, and those are not public API.

As an alternative you can do the same thing as you did for the SlowLog API where a filter can be supplied, and one thing the filter interface lets you do is specify a limit. 

I won't approve this without a limit option of some kind in the public user facing admin API.

Reply:
In the future please do not mark conversations that are not resolved as resolved.